tests/test_s3_storage.py

```diff
@@ -79,3 +79,28 @@ class TestStorage(PrivateS3Storage):
     storage = TestStorage()
 
     assert storage.get_path("test.txt") == "http://s3.fastapi.storages/test.txt"
+
+
+@mock_s3
+def test_s3_storage_duplicate_file_names(tmp_path: Path) -> None:
+    s3 = boto3.client("s3")
+    s3.create_bucket(Bucket="bucket")
+
+    tmp_file = tmp_path / "duplicate.txt"
+    tmp_file.write_bytes(b"123")
+
+    class TestStorage(PrivateS3Storage):
+        AWS_S3_CUSTOM_DOMAIN = "s3.fastapi.storages"
+
+    storage1 = TestStorage(overwrite_existing_files=False)
+    storage2 = TestStorage(overwrite_existing_files=False)
+    storage3 = TestStorage(overwrite_existing_files=False)
```

Comment:
You don't really need 3 storage instances. one would be ok.

Reply:
Then I need to assert after each write operation, it is one storage but three snapshots iof its state.